src/examples/common_plane_gpu.cpp

```diff
@@ -0,0 +1,280 @@
+// srun -n1 ./common_plane_ex --block1_res 4 4 4      --block1_min 0 0 0. --block1_max 1 1 1.05  --block2_res 4 4 4      --block2_min 0 0 0.95  --block2_max 1 1 2
+
+template <tribol::MemorySpace MSPACE, tribol::ExecutionMode EXEC>
+int runExample(const std::string& mesh_file, int num_contact_elems);
+
+/*!
+ * \brief Program main.
+ *
+ * This example runs the common plane + penalty algorithm for two opposing blocks
+ *
+ * \param [in] argc argument counter
+ * \param [in] argv argument vector
+ *
+ * \return rc return code, a non-zero return code indicates an error.
+ */
+int main( int argc, char** argv )
+{
+  ////////////////////////////////
+  //                            //
+  // SETUP THE EXAMPLE AND MESH //
+  //                            //
+  ////////////////////////////////
+
+  // initialize
+#ifdef TRIBOL_USE_MPI
+  MPI_Init( &argc, &argv );
+#endif
+  tribol::CommT problem_comm = TRIBOL_COMM_WORLD;
+  initialize_logger( problem_comm );
+
+#ifdef TRIBOL_USE_UMPIRE
+  umpire::ResourceManager::getInstance();  // initialize umpire's ResouceManager
+#endif
+
+  int err = 0;
+
+  for (int i{0}; i < 11; ++i)
+  {
+    int num_elems_1d = std::pow(2, i);
+    int num_contact_elems = std::pow(num_elems_1d, 2);
+    std::string mesh_file = TRIBOL_REPO_DIR "/data/single_layer_";
+    mesh_file = mesh_file + std::to_string(num_elems_1d) + ".mesh";
+    std::cout << "Running mesh file " << mesh_file << std::endl;
+#ifdef TRIBOL_USE_CUDA
+    auto mem_space = tribol::MemorySpace::Device;
+
+    if (mem_space == tribol::MemorySpace::Device)
+    {
+      err = runExample<tribol::MemorySpace::Device, tribol::ExecutionMode::Cuda>(mesh_file, num_contact_elems);
+    }
+    else
+#endif
+#ifdef TRIBOL_USE_HIP
+    auto mem_space = tribol::MemorySpace::Device;
+
+    if (mem_space == tribol::MemorySpace::Device)
+    {
+      err = runExample<tribol::MemorySpace::Device, tribol::ExecutionMode::Hip>(mesh_file, num_contact_elems);
+    }
+    else
+#endif
+    {
+#ifdef TRIBOL_USE_OPENMP
+      auto exec_mode = tribol::ExecutionMode::Sequential;
+      if (exec_mode == tribol::ExecutionMode::OpenMP)
+      {
+        err = runExample<tribol::MemorySpace::Host, tribol::ExecutionMode::OpenMP>(mesh_file, num_contact_elems);
+      }
+      else
+#endif
+      {
+        err = runExample<tribol::MemorySpace::Host, tribol::ExecutionMode::Sequential>(mesh_file, num_contact_elems);
+      }
+    }
+  }
+
+  axom::slic::flushStreams();
+  finalize_logger();
+
+#ifdef TRIBOL_USE_MPI
+  MPI_Finalize();
+#endif
+
+  return err;
+}
+
+template <tribol::MemorySpace MSPACE, tribol::ExecutionMode EXEC>
+int runExample(const std::string& mesh_file, int num_contact_elems)
+{
+
+  mfem::Device device;
+  switch (MSPACE)
+  {
+#ifdef TRIBOL_USE_CUDA
+    case tribol::MemorySpace::Device:
+      device.Configure("cuda");
+      break;
+#endif
+    default:
+// #ifdef TRIBOL_USE_OPENMP
```

Comment:
Should this be commented out, or should an error be reported?

Reply:
Fixed in updated example